Feature/coupon

hobby3530 opened the pull request:
Coupon 관련 Controller, Repository, Service 테스트 입니다.

hobby3530 committed:
[Refactor] CouponController 수정

@Valid, @RequestBody 추가

hobby3530 committed:
[Refactor] CouponMonthRepositoryImpl 수정

체크스타일 체크

hobby3530 committed:
[Refactor] Coupon관련 DTO 추가

hobby3530 committed:
[Refactor] Checkstyle 통과

hobby3530 committed:
[Refactor] MemberNotFoundException 추가

hobby3530 committed:
[Refactor] 기본 생성자 protected 제거

hobby3530 committed:
[Refactor] CouponPolicy DTO 수정

변수명 변경

hobby3530 committed:
[Refactor] CouponPolicy 엔티티 변경

변수명 변경

hobby3530 committed:
[Refactor] CouponTemplate DTO 변경

변수명 변경

hobby3530 committed:
[Refactor] File entity 변경

cascadeType 설정

hobby3530 committed:
[Refactor] CouponTemplateRepository QueryDsl 수정

QueryDsl 변경

hobby3530 committed:
[Refactor] CouponTemplateRepository QueryDsl 수정

QueryDsl 변경

hobby3530 committed:
[Refactor] service, repository 수정

변수명 변경에 따른 getter 변경

unhas01 committed:
[Test] Coupon 관련 테스트

controller, service, repository 테스트

hobby3530 committed:
[Test] Coupon 관련 테스트

controller, service, repository 테스트

HoFe-U commented on src/main/java/com/nhnacademy/bookpubshop/couponmonth/repository/impl/CouponMonthRepositoryImpl.java:
```diff
@@ -32,15 +32,15 @@ public Optional<GetCouponMonthResponseDto> getCouponMonth(Long monthNo) {
-                        .leftJoin(couponMonth.couponTemplate, file.couponTemplate)
-                .innerJoin(couponMonth.couponTemplate, couponTemplate)
-                .where(couponMonth.monthNo.eq(monthNo))
+        return Optional.ofNullable(from(couponMonth)
+                .join(couponMonth.couponTemplate, couponTemplate)
+                .leftJoin(file)
+                    .on(couponTemplate.templateNo.eq(file.couponTemplate.templateNo))
```
이곳에는 왜 Join 으로만 처리하셨나요?

hobby3530 replied:
innerjoin을 사용할 수도 있지만 join을 사용해도 innerjoin이 되는 것으로 알고 있습니다!

HoFe-U commented on src/main/java/com/nhnacademy/bookpubshop/coupontemplate/repository/impl/CouponTemplateRepositoryImpl.java:
```diff
@@ -48,26 +49,25 @@ public Optional<GetDetailCouponTemplateResponseDto> findDetailByTemplateNo(Long
+                .leftJoin(file).on(couponTemplate.templateNo.eq(file.couponTemplate.templateNo))
+                .join(couponTemplate.couponPolicy, couponPolicy)
+                .join(couponTemplate.couponType, couponType)
                 .leftJoin(couponTemplate.product, product)
                 .leftJoin(couponTemplate.category, category)
-                .leftJoin(couponTemplate.couponStateCode, couponStateCode)
+                .join(couponTemplate.couponStateCode, couponStateCode)
                 .select(Projections.constructor(GetDetailCouponTemplateResponseDto.class,
                         couponTemplate.templateNo,
-                        Projections.constructor(GetCouponPolicyResponseDto.class,
-                                couponPolicy.policyNo,
```
그냥 join 을 사용하신 이유가있을까요?

HoFe-U commented on src/main/java/com/nhnacademy/bookpubshop/file/entity/File.java:
```diff
@@ -45,7 +46,7 @@ public class File extends BaseCreateTimeEntity {
     @JoinColumn(name = "personal_inquiry_number")
     private PersonalInquiry personalInquiry;
 
-    @ManyToOne
+    @ManyToOne(cascade = CascadeType.ALL)
     @JoinColumn(name = "coupon_template_number")
     private CouponTemplate couponTemplate;
 
```
casecade 를 ALL 설정하신 이유를 알고싶습니다.

hobby3530 replied:
join시 에러 발생하여 이것저것 시도해보다가 넣은 것 같습니다. 삭제하도록 하겠습니다!

HoFe-U reviewed (approved):
수고하셨습니다.

seohyeon07 commented on src/test/java/com/nhnacademy/bookpubshop/coupon/controller/CouponControllerTest.java:
```diff
@@ -0,0 +1,220 @@
+import org.springframework.test.web.servlet.MockMvc;
+
+/**
+ * Some description here.
```
javadoc 작성해주세요

hobby3530 replied:
작성완료하였습니다!

seohyeon07 commented on src/test/java/com/nhnacademy/bookpubshop/coupon/service/CouponServiceTest.java:
```diff
@@ -0,0 +1,311 @@
+import org.springframework.test.util.ReflectionTestUtils;
+
+/**
+ * Some description here.
```
javadoc 추가해주세요

seohyeon07 commented on src/test/java/com/nhnacademy/bookpubshop/couponmonth/controller/CouponMonthControllerTest.java:
```diff
@@ -0,0 +1,322 @@
+import org.springframework.test.web.servlet.MockMvc;
+
+/**
+ * Some description here.
```
javadoc

seohyeon07 commented on src/test/java/com/nhnacademy/bookpubshop/couponmonth/service/CouponMonthServiceTest.java:
```diff
@@ -0,0 +1,251 @@
+import org.springframework.test.util.ReflectionTestUtils;
+
+/**
+ * Some description here.
```
javadoc

seohyeon07 reviewed (approved):
javadoc 확인 부탁드려요! 수고하셨습니다

hobby3530 committed:
[Docs] 쿠폰 관련 테스트 파일 자바독 수정

CouponControllerTest, CouponMonthControllerTest, CouponMonthServiceTest, CouponServiceTest 자바독 추가

hobby3530 committed:
[Refactor] 필요없는 cascade 삭제
